.github/workflows/reusable-pypi.yml

```
@@ -0,0 +1,38 @@
+on:
+  workflow_call:
+    inputs:
+      python_version:
+        required: false
+        default: '3.x'
+        type: string
+    secrets:
+      PYPI_TOKEN:
+        required: true
+jobs:
+  distribute-to-pypi:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "${{ inputs.python_version }}"
+
+      - name: Install Python Build
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install --upgrade build
+
+      - name: Build distribution
+        id: build
+        run: |
+          python -m build
+
+      - name: upload to PyPI.org
+        uses: pypa/gh-action-pypi-publish@v1.12.4
```

```suggestion
        uses: pypa/gh-action-pypi-publish@v1
```
or the current v1.13.0 if we prefer being specific